src/backend/InvenTree/order/models.py

```diff
@@ -638,10 +839,58 @@ def _action_complete(self, *args, **kwargs):
             trigger_event('purchaseorder.completed', id=self.pk)
 
     @transaction.atomic
-    def place_order(self):
+    def mark_order_pending(self, reason=''):
+        """Attempt to transition to PENDING status."""
+        return self.handle_transition(
+            self.status,
+            PurchaseOrderStatus.PENDING.value,
+            self,
+            self._action_pending,
+            reject_reason=reason,
+        )
+
+    @transaction.atomic
+    def request_approval(self, user=None):
+        """Attempt to transition to IN_APPROVAL status."""
+        return self.handle_transition(
+            self.status,
+            PurchaseOrderStatus.READY.value,
+            self,
+            self._action_request_approval,
+            user=user,
+        )
+
+    @transaction.atomic
+    def mark_order_ready(self, user):
+        """Attempt to transition to READY status."""
+        if self.requires_approval:
+            return self.handle_transition(
+                self.status,
+                PurchaseOrderStatus.READY,
+                self,
+                self._action_approve,
+                user=user,
+            )
+        elif self.can_be_ready:
+            return self.handle_transition(
+                self.status,
+                PurchaseOrderStatus.READY,
+                self,
+                self._action_ready,
+                user=user,
+            )
+        else:
+            raise ValidationError('Ready state setting not active')
+
+    @transaction.atomic
+    def place_order(self, user):
```

Comment:
This is breaking for plugin devs as a user is required and not optional

Reply:
Can make it optional. Will mean that placed_by won't be populated, but I presume that's okay for authors using that endpoint.

Reply:
I would prefer that, sourcegraph shows at least 20 source-available plugins using this function, probably more

Reply:
Consider it done 👌